tools/ci/src/commands/mod.rs

```diff
@@ -27,3 +31,99 @@ mod format;
 mod lints;
 mod test;
 mod test_check;
+
+#[derive(Debug, Default)]
+pub enum RustChannel {
+    #[default]
+    Stable,
+    Nightly,
+}
+
+fn status_to_result(status: ExitStatus) -> Result<(), ()> {
+    if status.success() {
+        Ok(())
+    } else {
+        Err(())
+    }
+}
+
+/// Runs a cargo command
+///
+/// stdout, stderr and stdin will be inherited.
+/// Args will be passed as is and no splitting will be performed (see [`args`](Command::args)).
+///
+/// Err(()) will be returned when:
+/// - There was a issue invoking the command (error message will be printed to stderr).
+/// - The command returned a non success exit code.
+pub fn run_cargo_command(
+    cargo_command: &str,
+    channel: RustChannel,
+    args: &[&str],
+    env: &[(&str, &str)],
+) -> Result<(), ()> {
+    let channel = match channel {
+        RustChannel::Stable => "stable",
+        RustChannel::Nightly => "nightly",
+    };
+
+    // We have to go through rustup as invoking cargo directly won't let us choose the toolchain.
+    // The +channel syntax you might be used to doesn't work as the cargo invoked by `Command` is
+    // different than the cargo you invoke in the console. Specifically your invocation goes through
+    // rustup which picks the correct cargo installation for you. `Command` seems to go directly to
+    // the system wide install of cargo.
+    Command::new("rustup")
+        .args(["run", channel, "cargo", cargo_command])
```

Comment:
I drive nightly Rust when developing Bevy and use the unstable configuration in `config_fast_builds.toml`. This broke that configuration by switching back to stable.

Would it be possible to fallback to the default toolchain, even if it is not necessarily stable? (It should be possible to query the configured toolchain for a given directory through `rustup show active-toolchain` or just `rustc --version`. You might need Regex for this, though.)

Given how this may be a more complex fix, I would be willing to split that out into a separate PR.

Reply:
6a125580e513e15768d6a09e90d274c25c0175f6 should fix your usecase.